lib/api/assertions/textEquals.js

```diff
@@ -0,0 +1,68 @@
+/**
+ * Check if an elements inner text equals the expected text
+ *
+ * @example
+ * 
+ * ```
+ *   this.demoTest = function (browser) {
+ *     browser.assert.textEquals('#main', 'The Night Watch');
+ *   };
+ * ```
+ * 
+ * @method textEquals 
+ * @param {string|object} definition The selector (CSS/Xpath) used to locate the element. Can either be a string or an object which specifies [element properties](https://nightwatchjs.org/guide#element-properties).
+ * @param {string} expected text to match text.
+ * @param {string} [msg] Optional log message to display in the output. If missing, one is displayed by default.
+ * @api assertions
+ */
+
+const {setElementSelectorProps} = require('../../utils');
+
+exports.assertion = function(definition, expected, msg) {
+  this.options = {
+    elementSelector: true
+  };
+ 
+  this.formatMessage = function() {
+    const message = msg || `Testing if element %s ${this.negate ? 'doesn\'t equal %s' : 'equals %s'}`;
+ 
+    return {
+      message,
+      args: [this.elementSelector, `'${expected}'`]
+    };
+  };
+ 
+  this.expected = function() {
+    return this.negate ? `doesn't equal '${expected}'` : `equals '${expected}'`;
+  };
+ 
+  this.evaluate = function(value) {
+    if (typeof value != 'string') {
+      return false;
+    }
+
+    return value === expected;
+  };
+
+  this.failure = function(result) {
+    return result === false || result && result.status === -1;
```

Comment:
We will drop support for `status` in future Nightwatch versions, so it's better if we can remove this condition

Reply:
@vaibhavsingh97 Ok, but what will be the commandResult if we remove the status for last test case of textEquals since status is -1 there.

Reply:
I don't think this has anything to do with the `status` being dropped at the user level. Even if we remove this we need its already using status over here : https://github.com/nightwatchjs/nightwatch/blob/6285ebcc8242147d9b47331a368e907d4bf94ba9/lib/api/assertions/_assertionInstance.js#L155